gusto/time_discretisation.py

```diff
@@ -1341,3 +1465,40 @@ def apply(self, x_out, *x_in):
             self.x[n].assign(x_in[n])
         solver.solve()
         x_out.assign(self.x_out)
+
+
+class ImplicitMidpoint(ImplicitMultistage):
+    u"""
+    Implements the Implicit Midpoint method as a 1-stage Runge Kutta method.
+
+    The method, for solving
+    ∂y/∂t = F(y), can be written as:
+
+    k0 = F[y^n + 0.5*dt*k0]
+    y^(n+1) = y^n + dt*k0
+    """
+    def __init__(self, domain, field_name=None, solver_parameters=None, limiter=None, options=None, butcher_matrix=None):
```

Comment:
I think `butcher_matrix` shouldn't be an argument here, since there is only one `butcher_matrix` for this scheme.

The same is true for the explicit schemes, e.g. `RK4`, `SSPRK3`, etc.